builtin/logical/pki/path_acme_directory.go

```diff
@@ -0,0 +1,138 @@
+}
+
+func patternAcmeDirectory(b *backend, pattern string) *framework.Path {
+	return &framework.Path{
+		Pattern: pattern,
+		Fields:  map[string]*framework.FieldSchema{},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback:                    b.acmeWrapper(b.acmeDirectoryHandler),
+				ForwardPerformanceSecondary: false,
+				ForwardPerformanceStandby:   true,
+			},
+		},
+
+		HelpSynopsis:    pathOcspHelpSyn,
+		HelpDescription: pathOcspHelpDesc,
+	}
+}
+
+type acmeOperation func(acmeCtx acmeContext, r *logical.Request, _ *framework.FieldData) (*logical.Response, error)
+
+type acmeContext struct {
+	baseUrl *url.URL
+	sc      *storageContext
+}
+
+func (b *backend) acmeWrapper(op acmeOperation) framework.OperationFunc {
+	return acmeErrorWrapper(func(ctx context.Context, r *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		sc := b.makeStorageContext(ctx, r.Storage)
+
+		if false {
+			// TODO sclark: Check if ACME is enable here
+			return nil, fmt.Errorf("ACME is disabled in configuration: %w", acme.ErrServerInternal)
+		}
+
+		baseUrl, err := getAcmeBaseUrl(sc, r.Path)
+		if err != nil {
+			return nil, err
+		}
+
+		acmeCtx := acmeContext{
+			baseUrl: baseUrl,
+			sc:      sc,
+		}
+
+		return op(acmeCtx, r, data)
+	})
+}
+
+func getAcmeBaseUrl(sc *storageContext, path string) (*url.URL, error) {
+	cfg, err := sc.getClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed loading cluster config: %w", err)
+	}
+
+	if cfg.Path == "" {
+		return nil, fmt.Errorf("ACME feature requires local cluster path configuration to be set: %w", acme.ErrServerInternal)
+	}
+
+	baseUrl, err := url.Parse(cfg.Path)
+	if err != nil {
+		return nil, fmt.Errorf("ACME feature a proper URL configured in local cluster path: %w", acme.ErrServerInternal)
+	}
+
+	directoryPrefix := ""
+	lastIndex := strings.LastIndex(path, "/acme/")
+	if lastIndex != -1 {
+		directoryPrefix = path[0:lastIndex]
+	}
+
+	return baseUrl.JoinPath(directoryPrefix), nil
+}
+
+func acmeErrorWrapper(op framework.OperationFunc) framework.OperationFunc {
+	return func(ctx context.Context, r *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		resp, err := op(ctx, r, data)
+		if err != nil {
+			return acme.TranslateError(err)
+		}
+
+		return resp, nil
+	}
+}
+
+func (b *backend) acmeDirectoryHandler(acmeCtx acmeContext, r *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	rawBody, err := json.Marshal(map[string]interface{}{
+		"newNonce":   acmeCtx.baseUrl.JoinPath("/acme/new-nonce").String(),
+		"newAccount": acmeCtx.baseUrl.JoinPath("/acme/new-account").String(),
+		"newOrder":   acmeCtx.baseUrl.JoinPath("/acme/new-order").String(),
+		"revokeCert": acmeCtx.baseUrl.JoinPath("/acme/revoke-cert").String(),
+		"keyChange":  acmeCtx.baseUrl.JoinPath("/acme/key-change").String(),
+		// Missing newAuthz on purpose until we support EAB
```

Comment:
This is correct but for the wrong reason. :-D 

We're not going to add support for `newAuthz` on initial pass, because it handles doing pre-authorization on an identifier before doing an order for it.

It lets you pre-prime the ACME responder to start doing challenge verifications before you submit the order, to get the certificate, so that you don't need to do challenges at order time.


We'll need a separate, non-ACME-specified path for getting EABs, which won't be an ACME path, but would be a regular Vault API response.